.github/workflows/scripts/comments.mjs

```diff
@@ -1,14 +1,70 @@
+import { readFile } from 'node:fs/promises'
+
+/**
+ * Posts the content of a diff as a comment on a Github issue
+ *
+ * @param {GithubActionContext} githubActionContext
+ * @param {number} issueNumber
+ * @param {DiffComment} diffComment
+ */
+export async function commentDiff (
+  githubActionContext,
+  issueNumber,
+  { path, title, markerText }
+) {
+  // Read diff from previous step
+  const diffText = await readFile(path, 'utf8')
+
+  // Add or update comment on PR
+  try {
+    await comment(
+      githubActionContext,
+      issueNumber,
+      markerText,
+      renderDiffBodyText(title, diffText)
+    )
+  } catch {
+    await comment(
+      githubActionContext,
+      issueNumber,
+      markerText,
+      // Unfortunately the best we can do here is a link to the "Artifacts"
+      // section as [the upload-artifact action doesn't provide the public
+      // URL](https://github.com/actions/upload-artifact/issues/50) :'(
+      `## ${title}\n` +
+        `The diff could not be posted as a comment. You can download it from the [workflow artifacts](${githubActionRunUrl(
+          githubActionContext.context.runId
+        )}#artifacts).`
+    )
+  }
+}
+
 /**
- * @param {object} githubContext - Github Action context
- * @param {import('@octokit/rest').Octokit} githubContext.github
- * @param {import('@actions/github').context} githubContext.context
- * @param {number} issueNumber - GitHub issue or PR number
+ * @param {string} title - The title for the comment
+ * @param {string} diffText - The diff content
+ * @returns {string} The content for the comment body
+ */
+function renderDiffBodyText (title, diffText) {
+  if (diffText.length) {
+    return `## ${title}\n` +
+      '```diff\n' +
+      `${diffText}\n` +
+      '```'
+  } else {
+    return `## ${title}\nThe diff is empty.`
+  }
+}
+
+/**
+ * @param {GithubActionContext} githubContext - Github Action context
+ * @param {number} issueNumber - The number of the issue/PR on which to post the comment
  * @param {string} markerText - A unique marker used to identify the correct comment
  * @param {string} bodyText - The body of the comment
  */
-export async function comment ({ github, context }, issueNumber, markerText, bodyText) {
+export async function comment ({ github, context, commit }, issueNumber, markerText, bodyText) {
```

Comment:
I quite like the obvious params:

```js
function ({ github, context, commit }, issueNumber, markerText, bodyText) {}
```

But if we're adopting objects on [#L13](https://github.com/alphagov/govuk-frontend/pull/3671/files#diff-e831ad02a0354c8c4b18febab2f20c25a2baa97f742c13b75499020dd29d58c9R13) let's be consistent:

```js
function ({ github, context, commit }, issueNumber, { title, bodyText }) {}
```